src/controllers/post.controller.ts

```diff
@@ -16,6 +16,17 @@ class PostController {
       next(error);
     }
   };
-}
 
-export default PostController;
+  public updatePost = async (req: RequestWithUser, res: Response, next: NextFunction) => {
+    try {
+      const userId = req.user.id
+      const updatedPostData: BasePostDto = req.body;
```

Comment:
I think we should have a `UpdatePostDTO` that maybe looks like this
`type UpdatePostDTO = {
  id: number,
  content?: string;
  media?: string;
  categoryName?: string;
}`
This way, we don't care about passing in the userId.